coffeecard/CoffeeCard.WebApi/Services/LoginLimiter.cs

```diff
@@ -0,0 +1,39 @@
+using System;
+using System.Collections;
+using System.Collections.Concurrent;
+
+namespace CoffeeCard.WebApi.Services
+{
+    public class LoginLimiter : ILoginLimiter
+    {
+        private ConcurrentDictionary<string, (DateTime, int)> _loginAttempts;
+
+        public LoginLimiter()
+        {
+            _loginAttempts = new ConcurrentDictionary<string, (DateTime, int)>();
+        }
+
+        /// <summary>
```

Comment:
Since the documentation in this instance is referring to the implementation I would think not? Otherwise, we are all of a sudden saying all potential implementations is implemented this way :/

Reply:
I can see what you mean; let me rephrase :slightly_smiling_face: . For the most cases, the caller will use the interface and not call the implementation, so provide documentation for the caller, it would be most useful to add the documentation to the interface. However as you're also pointing out, your code comments address the implementation, so maybe a way to go about it, is to have the implementation specific comments on the implementation and the rest on the interface :slightly_smiling_face:

Reply:
However, since the current level of documented methods, is really low, I would consider this change voluntary :blush: